source/scripts/Background/controllers/transactions/evm.ts

```diff
@@ -0,0 +1,147 @@
+      apiUrl: networkApiUrl,
+    } = activeNetwork;
+
+    const etherscanSupportedNetworks = [
+      'homestead',
+      'ropsten',
+      'rinkeby',
+      'goerli',
+      'kovan',
+    ];
+
+    const networkByLabel =
+      networkChainId === 1 ? 'homestead' : networkLabel.toLocaleLowerCase();
+
+    if (isNetworkDefault) {
+      if (etherscanSupportedNetworks.includes(networkByLabel)) {
+        const etherscanProvider = new ethers.providers.EtherscanProvider(
+          networkByLabel,
+          'K46SB2PK5E3T6TZC81V1VK61EFQGMU49KA'
+        );
+
+        const getTxHistory = await etherscanProvider.getHistory(userAddress);
+
+        const limitedTxHistory = getTxHistory.slice(0, 30); // Limit last 30 TXs
+
+        const formattedTxs: ITransactionResponse[] = await Promise.all(
+          limitedTxHistory.map(
+            async (tx) =>
+              await getFormattedTransactionResponse(etherscanProvider, tx)
+          )
+        );
+
+        return formattedTxs;
+      }
+
+      const query = `?module=account&action=txlist&address=${userAddress}`;
+
+      const {
+        data: { result },
+      } = await axios.get(`${networkApiUrl}${query}`);
+
+      if (!isString(result)) {
+        const provider = new ethers.providers.JsonRpcProvider(networkUrl);
+
+        const formattedTxs: ITransactionResponse[] = await Promise.all(
+          result.map(
+            async (tx: ITransactionResponse) =>
+              await getFormattedTransactionResponse(provider, tx)
+          )
+        );
+
+        return formattedTxs;
+      }
+    }
+  };
+
+  const getSocketPendingTransactions = async (): Promise<IEvmTransaction[]> => {
+    const { activeNetwork, accounts, activeAccount } = store.getState().vault;
+
+    const { address: userAddress, transactions: userTransactions } =
+      accounts[activeAccount];
+
+    const getChain = chains.get(activeNetwork.chainId) as Chain;
+
+    const wssUrlByChain = getChain.rpc.find((rpc) => rpc.startsWith('wss://'));
+
+    const wssNeedApiKey = Boolean(wssUrlByChain.includes('API_KEY'));
+
+    const validatedUrl = wssNeedApiKey ? null : wssUrlByChain;
+
+    if (isNil(validatedUrl)) return [];
+
+    const wssProvider = new ethers.providers.WebSocketProvider(validatedUrl);
+
+    wssProvider.on('error', (error) => {
+      console.log('error socket transactions', error);
+      return userTransactions.ethereum as IEvmTransaction[];
+    });
+
+    wssProvider.on('pending', async (txHash: string) => {
+      const transaction = await wssProvider.getTransaction(txHash);
+
+      const { from, to, hash, blockNumber } = transaction;
+
+      const isValidTx =
+        transaction && (from === userAddress || to === userAddress);
+
+      if (isValidTx) {
+        const { timestamp } = await wssProvider.getBlock(Number(blockNumber));
+
+        const txAlreadyExists =
+          userTransactions.ethereum.findIndex(
+            (transaction: ITransactionResponse) => transaction.hash === hash
```

Comment:
'transaction' is already declared in the upper scope on line 108 column 13 (this might generate scope errors)
suggestion: replace one of the `transaction` const names